q2_moshpit/eggnog/_utils.py

```diff
@@ -26,3 +39,112 @@ def _parse_build_diamond_db_params(arg_key, arg_val) -> List[str]:
         return [f"--{arg_key}"]
     else:
         return [f"--{arg_key}", str(arg_val)]
+
+
+def _download_and_build_hmm_db(taxon_id):
+    pressed_hmm_db_obj = PressedProfileHmmsDirectoryFmt()
+    hmm_db_obj = ProteinMultipleProfileHmmDirectoryFmt()
+    idmap_obj = EggnogHmmerIdmapDirectoryFmt()
+
+    with tempfile.TemporaryDirectory() as tmp:
+        _try_wget(
+            f"{tmp}/{taxon_id}_hmms.tar.gz",
+            "http://eggnog5.embl.de/download/eggnog_5.0/per_tax_level/"
+            f"{taxon_id}/{taxon_id}_hmms.tar.gz",
+            "Error during HMMER database download"
+        )
```

Comment:
What I did in order to prevent this is that the first thing that happens is that the taxon id is validated. So this should never happen as long as `_download_and_build_hmm_db` is never called directly. If you try to run the action specifying 8 as taxon id you would get 

```
'8' is not valid taxon ID. To view all valid taxon IDs inspect e5.taxid_info.tsv. You can download it with this command: wget http://eggnog5.embl.de/download/eggnog_5.0/e5.taxid_info.tsv
```

Reply:
Well, clearly it does happen. First of all, what your function checks is whether the taxon ID is present in the taxid_info file (which it should!) - what I am referring to is those being present/absent on the server. It's great it works with taxon 8 but that's not the taxon I used - mine was `1773` (in which case, your message is even more confusing: what does the 8 even mean there?). Please build in exception handling at the download level to avoid this kind of situation.

Reply:
Oh, I see what you mean now. My mistake. I thought if it was on the taxid_info file it's on the FTP server, but that is not true. 8 is the return code from the failed wget command, which apparently means "Server issued an error response", which is funny enough not so much more informative than the code itself 😅. I will...

1. Change the error message so it says, "error code: x"
2. change the validation so it actually checks whether the taxon id is on the server rather than on the taxid_info file.

Reply:
Hmm, I wonder for point 2: maybe you should actually do both, just to be sure that the id is present everywhere where it should be...

Reply:
Yeah, that sounds right 🙌